pkg/server/server_http.go

```diff
@@ -54,6 +55,21 @@ func newHTTPServer(
 	}
 }
 
+// HSTSEnabled is a boolean that enables HSTS headers on the HTTP
+// server. These instruct a valid user agent to use HTTPS *only*
+// for all future connections to this host.
+var HSTSEnabled = func() *settings.BoolSetting {
+	s := settings.RegisterBoolSetting(
+		settings.TenantWritable,
+		"security.hsts.enabled",
```

Comment:
Once this code is checked in, is this config turned via start command or SQL command? Thx

Reply:
@dhartunian can you help with the above question ^ ? Thx

Reply:
Once this is checked-in, you can enable/disable via SQL:

```
set cluster setting security.hsts.enabled = true;
```

or 

```
set cluster setting security.hsts.enabled = false;
```

Reply:
if you would like to use the cluster setting I would advise waiting until we merge this so the naming of the setting is settled. otherwise future upgrades could create regressions.

Reply:
Yeah, sure, I will wait for your PR to merge. Then I should be able to pull the latest version of `v21.1` (we are currently at v21.1.12). Thx

Reply:
Ah, if you need this on `21.1`, I will need to backport the change. I will set the labels on this PR so that this gets done. 

Once we merge to `master` that will get it into the upcoming `22.1` release. Any feature that needs to be present on `21.1` or `21.2` will need to be backported.

Reply:
@dhartunian I see, please do backport to `21.1`. We intentionally keep one version lower than the current GA one, assuming new versions need some time to be mature. When `22.1` becomes GA, we may want to go to `21.2`. But for now, we will use `21.1`. Thx for the quick turnaround!